packages/mcp-server-supabase/src/server.ts

```diff
@@ -328,6 +329,47 @@ export function createSupabaseMcpServer(options: SupabaseMcpServerOptions) {
           return await executeSql(project_id, query);
         },
       }),
+      get_logs: tool({
+        description:
+          'Gets logs for a Supabase project by service type. Use this to help debug problems with your app. This will only return logs within the last minute. If the logs you are looking for are older than 1 minute, re-run your test to reproduce them.',
+        parameters: z.object({
+          project_id: z.string(),
+          service: z
+            .enum([
+              'api',
+              'postgres',
+              'edge-function',
+              'auth',
+              'storage',
+              'realtime',
+            ])
+            .describe('The service to fetch logs for'),
+        }),
+        execute: async ({ project_id, service }) => {
+          const now = Date.now();
+          const response = await managementApiClient.GET(
+            '/v1/projects/{ref}/analytics/endpoints/logs.all',
+            {
+              params: {
+                path: {
+                  ref: project_id,
+                },
+                query: {
+                  // Limit to the last minute
+                  iso_timestamp_start: new Date(now - 60 * 1000).toISOString(),
+                  iso_timestamp_end: new Date(now).toISOString(),
```

Comment:
```suggestion
                  iso_timestamp_end: new Date(now + 5 * 60 * 1000).toISOString(),
```

Shall we set the end time to 5 minutes in the future to account for any local clock drift?

Reply:
Or perhaps we can omit both timestamps entirely since the default is last 1 minute.

https://github.com/orgs/supabase/discussions/34634